core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1195,15 +1188,20 @@ private boolean checkConsistencyInternal(LockedInodePath inodePath) throws Inval
       if (ufs == null) {
         return true;
       }
-      if (!inode.isPersisted()) {
-        return !ufs.exists(ufsPath);
+      UfsStatus ufsStatus;
+      try {
+        ufsStatus = ufs.getStatus(ufsPath);
+      } catch (FileNotFoundException e) {
+        throw new RuntimeException(e);
```

Comment:
return false here and add back `inode.isPersisted` instead of throwing an exception